pkgs/swift2objc/lib/src/transformer/transform.dart

```diff
@@ -52,13 +51,8 @@ Declaration transformDeclaration(
   }
 
   return switch (declaration) {
-    ClassDeclaration() => transformClass(
-        declaration,
-        globalNamer,
-        transformationMap,
-      ),
-    StructDeclaration() => transformStruct(
-        declaration,
+    ClassDeclaration() || StructDeclaration() => transformCompound(
+        declaration as CompoundDeclaration,
```

Comment:
Do you need this `as`? I would have thought this would work automatically since they're both subtypes of `CompoundDeclaration`?

Reply:
Yup I thought so too, but for some reason it's not detected.